sql/wsrep_sst.cc

```diff
@@ -1276,6 +1414,78 @@ static int sst_donate_other (const char*   method,
   return arg.err;
 }
 
+/*
+  Validate SST request string.
+  The protocol is: method\0data\0
+
+  For xtrabackup-v2 it is: method\0addresspath\0
+  like:
+  "xtrabackup-v2\000127.0.0.1:4444/xtrabackup_sst//1"
+*/
+static bool is_sst_request_valid(const std::string &msg)
+{
+  size_t method_len= strlen(msg.c_str());
+
+  if (method_len == 0)
+  {
+    return false;
+  }
+
+  std::string method= msg.substr(0, method_len);
+
+  // Is this method allowed?
+  std::vector<std::string>::iterator res= std::find(
+      allowed_sst_methods.begin(), allowed_sst_methods.end(), method);
+  if (res == allowed_sst_methods.end())
+  {
+    return false;
```

Comment:
Maybe an error message here describing how to fix this would be good (saying that they need to add the method to the list of allowed methods). Otherwise, they wouldn't know how to fix the error.  Everything else looks good.

Reply:
In case of rejection, the caller function writes the error message "Invalid sst_request:" and dumps the whole request string. I don't think we should add more hints about what is wrong. We need to document the new parameter anyway. And honestly, I don't think there are no external users having their own sst scripts :)